src/libos/src/process/do_spawn/exec_loader.rs

```diff
@@ -1,21 +1,22 @@
+use super::super::elf_file::*;
 use super::ThreadRef;
 use crate::fs::{FileMode, INodeExt};
 use crate::prelude::*;
 use rcore_fs::vfs::INode;
 use std::ffi::CString;
 
-/// Load an ELF file itself or a script's interpreter into a vector.
-///
-/// If the file is an executable binary, then just load this file.
-/// If the file is an script text, then parse the shebang and load
-/// the interpreter.
-pub fn load_exec_file_to_vec(
+pub fn load_exec_file_hdr_to_vec(
     file_path: &str,
     current_ref: &ThreadRef,
-) -> Result<(Option<String>, Vec<u8>)> {
-    let file_buf = load_file_to_vec(&file_path, current_ref)?;
-    let is_script = is_script_file(&file_buf);
-    if is_script {
+) -> Result<(Option<String>, Arc<dyn INode>, Vec<u8>, ElfHeader)> {
+    let (inode, file_buf, elf_hdr) = load_file_hdr_to_vec(&file_path, current_ref)?;
+    if elf_hdr.is_some() {
+        Ok((None, inode, file_buf, elf_hdr.unwrap()))
+    } else {
+        // loaded file is not Elf format, try script file
+        if !is_script_file(&file_buf) {
```

Comment:
Is the file_buf correctly initialized in this case?

Reply:
file_buf is always initialized in `read_elf64_as_vec` in deps/sefs/rcore-fs/src/vfs.rs.